beacon/goclient/proposer.go

```diff
@@ -38,16 +45,49 @@ func (gc *goClient) GetBeaconBlock(slot phase0.Slot, committeeIndex phase0.Commi
 	case spec.DataVersionAltair:
 		return beaconBlock.Altair, beaconBlock.Version, nil
 	case spec.DataVersionBellatrix:
+		if b := beaconBlock.Bellatrix.Body; b != nil && b.ExecutionPayload != nil {
+			gc.logBlock(slot, b.ExecutionPayload.BlockHash, beaconBlock.Version, false)
+		}
 		return beaconBlock.Bellatrix, beaconBlock.Version, nil
 	case spec.DataVersionCapella:
+		if b := beaconBlock.Capella.Body; b != nil && b.ExecutionPayload != nil {
+			gc.logBlock(slot, b.ExecutionPayload.BlockHash, beaconBlock.Version, false)
+		}
 		return beaconBlock.Capella, beaconBlock.Version, nil
 	default:
-		return nil, DataVersionNil, errors.New(fmt.Sprintf("beacon block version %s not supported", beaconBlock.Version))
+		return nil, DataVersionNil, fmt.Errorf("beacon block version %s not supported", beaconBlock.Version)
 	}
 }
 
-func (gc *goClient) GetBlindedBeaconBlock(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, graffiti, randao []byte) (ssz.Marshaler, spec.DataVersion, error) {
-	return nil, DataVersionNil, nil
+func (gc *goClient) GetBlindedBeaconBlock(slot phase0.Slot, graffiti, randao []byte) (ssz.Marshaler, spec.DataVersion, error) {
+	sig := phase0.BLSSignature{}
+	copy(sig[:], randao[:])
+
+	reqStart := time.Now()
+	beaconBlock, err := gc.client.BlindedBeaconBlockProposal(gc.ctx, slot, sig, graffiti)
+	if err != nil {
+		return nil, 0, err
+	}
+	metricsProposerDataRequest.Observe(time.Since(reqStart).Seconds())
+
+	if beaconBlock == nil {
+		return nil, 0, fmt.Errorf("block is nil")
+	}
+
+	switch beaconBlock.Version {
+	case spec.DataVersionBellatrix:
+		if b := beaconBlock.Bellatrix.Body; b != nil && b.ExecutionPayloadHeader != nil {
```

Comment:
if `Body` is `nil` shouldn't we return an error?

Reply:
I didn't find any cases when it's `nil` when I did a quick look, but let's log it

Reply:
if you didn't find any cases then I'm not sure why we have this check?

Reply:
I'm not completely sure if it's never `nil` and if it is, this will panic. Also, it would trigger the linter (dereferencing a pointer without checking if it's `nil` is indeed bad).

But I agree we better log this, despite the code would look bulkier.